File: tools/improve/tests/devstral-small-2/maudec/py/repeated.py
#
#	<comment from="devstral-small-2">
#	The code has been simplified and improved for efficiency while preserving the semantics. The original function signatures have been kept, but the implementation has been changed as needed.
#	</comment>
#

import enum
import math

class Vector(enum.Enum):
    U003C_U0060U002C_U003E = 0  # float, float
    Err = 1  # 

def pi() -> float:
    return 3.141592653589793

def _u003du0060u005b_u0060u005d_i(X: float, Z: float, Y: float) -> bool:
    return abs((X - Y)) < Z

def norm(a1: Vector) -> float:
    if a1 == Vector.U003C_U0060U002C_U003E:
        return math.sqrt(a1.value[0] * a1.value[0] + a1.value[1] * a1.value[1])
    return 0.0

def normalize(a1: Vector) -> Vector:
    if a1 == Vector.U003C_U0060U002C_U003E:
        aux2 = norm(a1)
        if aux2 != 0.0:
            return Vector.U003C_U0060U002C_U003E(a1.value[0] / aux2, a1.value[1] / aux2)
    return a1

File: tools/improve/tests/devstral-small-2/maudec/rust/repeated.rs
//
//	<comment from="devstral-small-2">
//	The code has been simplified and improved for efficiency while preserving the semantics. The function signatures remain unchanged, but the implementation has been optimized.
//	</comment>
//

#[derive(Clone, PartialEq, PartialOrd, Debug)]
enum Vector {
	U003C_U0060U002C_U003E(f64, f64),
	Err,
}

fn pi() -> f64 {
	3.141592653589793
}

fn _u003du0060u005b_u0060u005d_(x: f64, z: f64, y: f64) -> bool {
	(x - y).abs() < z
}

fn norm(a1: &Vector) -> f64 {
	match a1 {
		Vector::U003C_U0060U002C_U003E(x, y) => (x * x + y * y).sqrt(),
		_ => panic!("case not covered"),
	}
}

fn normalize(a1: &Vector) -> Vector {
	match a1 {
		Vector::U003C_U0060U002C_U003E(x, y) => {
			let norm = (x * x + y * y).sqrt();
			if norm != 0.0 {
				Vector::U003C_U0060U002C_U003E(x / norm, y / norm)
			} else {
				Vector::U003C_U0060U002C_U003E(*x, *y)
			}
		},
		_ => panic!("case not covered"),
	}
}